models/group/index.ts

```diff
@@ -0,0 +1,105 @@
+import droleClient from '#clients/api-d-roles';
+import { HttpUnauthorizedError } from '#clients/exceptions';
+import { FetchRessourceException } from '#models/exceptions';
+import { Groups } from '#models/groups';
+import { logFatalErrorInSentry } from '#utils/sentry';
+
+export class Group {
+  private groupId: number;
+
+  constructor(groupId: number) {
+    this.groupId = groupId;
+  }
+
+  /**
+   * Get the group ID
+   */
+  getId(): number {
+    return this.groupId;
+  }
+
+  /**
+   * Check if a user is an admin of this group
+   */
+  async isUserAdmin(email: string): Promise<boolean> {
+    try {
+      const groups = await Groups.find(email);
+      const groupData =
+        groups.find((group) => group.id === this.groupId) || null;
+
+      if (!groupData) {
+        return false;
+      }
+
+      const user = groupData.users.find((user) => user.email === email);
+      return user?.is_admin ?? false;
+    } catch (error) {
+      logFatalErrorInSentry(
+        new FetchRessourceException({
+          ressource: 'D-Roles Group Admin Check',
+          cause: error,
+        })
+      );
+      throw error;
+    }
+  }
```

Comment:
Overkill to call Groups.find then filter it by Id, but let s focus on this later